### PR TITLE
Upgrade to tf 12 syntax; add terratest plumbing

### PR DESCRIPTION
**Pull request by jsclarridge**
Per https://www.pivotaltracker.com/story/show/169407344, update the module to use Terraform 12 syntax and introduce Terratest into the module. The PR also includes:

- Makefile - For running pre-commit and Terratest
- golangci-lint - Go linting
- dependabot - To keep go modules up to date
- README - Updates for 0.12 versioning and how to run tests

The Terratest example lives in examples/simple, which instantiates this module with the default settings. Running against our CI AWS account returns

--- PASS: TestTerraformAwsLogs (41.80s)
PASS
ok  	github.com/trussworks/terraform-aws-logs/test	41.815s

### CODE REVIEW

File: .terraform-version [hunk 1]
```diff
@@ -1 +0,0 @@
-0.11.14
```
Comment (chrisgilmerproj) [inline .terraform-version]: Would it be better to update this to `0.12.12` or `0.12` or is the thinking that we shouldn't preference a version for `tfenv` in our repos?

File: README.md [hunk 2]
```diff
@@ -81,28 +80,35 @@ Logging from the following services is supported for both cases:
       ])
     }
 
+## Terraform Versions
+
+Terraform 0.12. Pin module version to ~> 4.x Submit pull-requests to master branch.
+
+Terraform 0.11. Pin module version to ~> 3.5.0. Submit pull-requests to terraform011 branch.
```
Comment (chrisgilmerproj) [inline README.md]: For some reason I thought 3.5.0 was already tf12. But I'm ok with this.

Comment (jsclarridge) [reply README.md]: The module currently supports both. v.3.4.0 added support for tf 11 again.

File: .terraform-version (hunk 1 above)
Comment (jsclarridge) [reply .terraform-version]: As we've migrated some of the other modules to TF 12, we've added a versions.tf file (also in this PR) which constrains the major version. I think that makes sense as a more tool-agnostic approach. `tfenv` and specific `.terraform-version` files can still be managed at the project level.

Comment (chrisgilmerproj) [reply .terraform-version]: That's fair. I've only found it useful when using `terraform 0.12upgrade` in this repo. But generally I do work from other repos. So I guess its up to you.

**Pull request merged**